### ISSUE TITLE
no tofu token but not in correct session state

### ISSUE THREAD

**Issue by EasyNV** [hashicorp/boundary#2414]
**Describe the bug**
Hello, I tried to use boundary to connect to the postgresql server using this command
```
boundary connect -target-id $target_id -listen-port 5432
```

And when I tried to connect by using `psql` client it will show this
```
Proxy listening information:
  Address:             127.0.0.1
  Connection Limit:    -1
  Expiration:          Wed, 31 Aug 2022 20:21:38 WIB
  Port:                5432
  Protocol:            tcp
  Session ID:          s_plkWXBdEe6

  Credentials:
    Credential Source ID:   clvlt_LLQxVJDeQv
    Credential Source Name: Engineer Credentials
    Credential Store ID:    csvlt_GUOLnIH4y5
    Credential Store Type:  vault
    Secret:
        {
              "password": "NPfKcYcXsgIxPjS7Ta4-",
              "username": "v-token-v-dev-en-qlZFSfgRGrNbv6kceQOy-1661923298"
        }

error reading handshake result: failed to read protobuf message: failed to get reader: received close frame: status = StatusInternalError and reason = "refusing to activate session"
```

Currently I use this version of boundary on the client, server and worker
```
boundary version

Version information:
  Git Revision:        d9eba38993eb70820a396894f2f1e28601d13c3d
  Version Number:      0.10.3
```

**To Reproduce**
1. Run `boundary connect -target-id $target_id -listen-port 5432`
2. Run `psql -h 127.0.0.1 -d postgres -U $boundary_token_username`
3. See error

**Expected behavior**
The boundary agent is able to serve the connections. This worked fine with version 0.9.1

[controller-1.log](https://github.com/hashicorp/boundary/files/9458689/controller-1.log)
[controller-2.log](https://github.com/hashicorp/boundary/files/9458690/controller-2.log)
[worker.log](https://github.com/hashicorp/boundary/files/9458691/worker.log)

**Comment by EasyNV** [comment]
I see that v0.10.2 should have fixed the issue as mentioned here https://github.com/hashicorp/boundary/issues/2362.

But somehow I still got the issue in the v0.10.3 😕

**Comment by matheusmgon** [comment]
I see that error on try connection in SSH too :/

The problem with migration was fixed ( when trying to upgrade the version 0.10.1 to 0.10.2), but the error when trying to **use the same session to connect multiple times** did not occur well yet :/

![boundary-screenshot-error-bd](https://user-images.githubusercontent.com/33628133/187635583-893b7c09-c423-4a65-89a6-ba52e048e505.png)
![boundary-screenshot-error-ssh](https://user-images.githubusercontent.com/33628133/187635592-a480aedc-1510-4dcd-a8d8-64d564c6f999.png)


PS: Yeah, I configured on the target to not limit multiple connections, and I have one worker for each environment (on the version 0.8 ~ 0.9 worked fine)

**Comment by ap-vishal** [comment]
Also getting the same error even after upgrading to 0.10.3 but it is only happening when the worker count is more than 1, if we have just 1 worker then the error is not happening, which was the case with 0.10.2 or 0.10.1.

**Comment by irenarindos** [comment]
Hi all- thank you so much for reporting this. The fix for this issue (#2369) will be included in 10.4
